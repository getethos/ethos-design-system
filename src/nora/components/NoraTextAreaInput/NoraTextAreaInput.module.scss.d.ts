// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'NoraTextInput': string;
}
export const cssExports: CssExports;
export default cssExports;
