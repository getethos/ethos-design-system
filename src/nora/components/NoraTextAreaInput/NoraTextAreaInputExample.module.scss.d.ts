// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'LabelClassExample': string;
  'TextClassExample': string;
}
export const cssExports: CssExports;
export default cssExports;
