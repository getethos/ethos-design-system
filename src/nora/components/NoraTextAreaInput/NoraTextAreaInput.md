Nora Text Area Input (Default)

```jsx
<NoraTextAreaInput
  data-tid="nora-textarea-input"
  name="default-example"
  placeholder="Enter text here"
/>
```

Nora Text Area Input with Formatting

```jsx
import styles from './NoraTextAreaInputExample.module.scss';
<NoraTextAreaInput
  data-tid="nora-textarea-input"
  name="formatted-example"
  labelCopy="OPTIONAL LABEL"
  labelClassName={styles.LabelClassExample}
  textClassName={styles.TextClassExample}
  placeholder="Enter text here"
/>
```
