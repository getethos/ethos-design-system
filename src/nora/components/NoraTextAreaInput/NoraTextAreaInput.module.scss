@import '../Attributes.css';

.NoraTextInput {
  width: 100%;
  box-sizing: border-box;
  background-color: var(--White);
  border-style: solid;
  border-width: 1px;
  border-color: var(--NoraNeutral-200);
  border-radius: var(--Radius-3);
  padding: 9px var(--Space-16) 9px;
  color: var(--NoraNeutral-400);
  transition: border-color 250ms ease;
  outline: 0;
  margin: 0;
}