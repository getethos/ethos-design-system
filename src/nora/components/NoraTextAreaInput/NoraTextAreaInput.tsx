import * as React from 'react'
import { useEffect, useState } from 'react'
import { PublicFormProps } from '../../../components/Form/Form'
import { BaseTextAreaInput } from '../../../components/TextAreaInput/BaseTextAreaInput'
import useErrorMessage from '../../../hooks/useErrorMessage'
import useInputValidation from '../../../hooks/useInputValidation'
import styles from './NoraTextAreaInput.module.scss'
type NoraTextAreaInputProps = PublicFormProps & {
  disabled?: boolean
  value?: string
  labelCopy?: string
  labelClassName?: string
  textClassName?: string
  placeholder?: string
  onBlur?: (...args: any[]) => any
  onFocus?: (...args: any[]) => any
  onChange?: (...args: any[]) => any
  initialValue?: any
}
export const NoraTextAreaInput: React.SFC<NoraTextAreaInputProps> = ({
  disabled,
  name,
  value,
  labelCopy,
  labelClassName,
  textClassName,
  placeholder,
  onBlur,
  onFocus,
  onChange,
  formChangeHandler,
  validator,
  initialValue,
  currentValue,
  currentError,
  setFieldTouched,
  ...rest
}) => {
  const classes = [styles.NoraTextInput, textClassName].join(' ')
  // Set up validation hooks
  const [getError, setError, , validate] = useErrorMessage(validator)
  const [val, setVal] = useState(currentValue || initialValue || '')
  const [touched, setTouched] = useState(value || initialValue ? true : false)
  const [doValidation] = useInputValidation({
    validate,
    setError,
    formChangeHandler,
  })
  const handleChange = (ev) => {
    const val = ev.target.value
    setVal(val)
    // We call setTouched in onBlur, so can reliably call getter here
    doValidation(val, touched)
    if (onChange) onChange(ev)
  }
  const setAllTouched = () => {
    // We set touched to change the react state, but it's async and
    // processing still, so, we use a flag for doValidation
    setTouched(true)
    // Also tell the form we've been touched
    if (setFieldTouched) {
      setFieldTouched(true)
    }
  }
  /**
   * If consumer's used the brute force `value` prop, we prioritize that
   * (this is similar to what we're doing in CheckboxInput for `checked`)
   */
  const resolvedValue = typeof value !== 'undefined' ? value : val
  // Initial Value aka prefilledâ€”are considered "touched", but must prevalidate
  // which will in turn update the internal form state as to their validity
  useEffect(() => {
    if (!!formChangeHandler && (resolvedValue || initialValue)) {
      doValidation(resolvedValue || initialValue, '')
    }
  }, [])
  const handleBlur = (ev) => {
    setAllTouched()
    doValidation(ev.target.value, true)
    if (onBlur) onBlur(ev)
  }
  // TODO: this indicates a field level error on the NoraTextAreaInput
  if (getError(currentError, touched)) {
    // console.log('NoraTextAreaInput field error: ', currentError)
  }
  return (
    <>
      {labelCopy && <label className={labelClassName}>{labelCopy}</label>}
      <BaseTextAreaInput
        className={classes}
        disabled={disabled}
        name={name}
        placeholder={placeholder}
        onBlur={handleBlur}
        onFocus={onFocus}
        onChange={handleChange}
        value={resolvedValue}
        data-tid={rest['data-tid']}
      />
    </>
  )
}
