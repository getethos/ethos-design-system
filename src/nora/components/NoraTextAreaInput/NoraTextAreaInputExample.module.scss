@import '../Attributes.css';

.LabelClassExample {
  font-weight: 500;
  font-size: var(--Nora-Font-11);
  line-height: 16px;
  color: var(--NoraNeutral-400);
}

.TextClassExample {
  margin-top: var(--Space-12);
  font-size: calc(var(--compact-font-size)* 1px);
  line-height: 20px;
  height: 100px;
}