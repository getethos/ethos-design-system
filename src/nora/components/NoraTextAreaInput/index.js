export { NoraTextAreaInput } from './NoraTextAreaInput.js'
