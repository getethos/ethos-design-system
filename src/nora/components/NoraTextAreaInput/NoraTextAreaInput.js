import React from 'react'
import PropTypes from 'prop-types'
import { TextAreaInput } from '../../../components/index'
import styles from './NoraTextAreaInput.module.scss'

export const NoraTextAreaInput = ({
  disabled,
  name,
  value,
  labelCopy,
  labelClassName,
  textClassName,
  placeholder,
  onBlur,
  onFocus,
  onChange,
  ...rest
}) => {
  return (
    <>
      {labelCopy && <p className={labelClassName}>{labelCopy}</p>}
      <BaseTextAreaInput
        className={textClassName}
        disabled={disabled}
        name={name}
        placeholder={placeholder}
        onPaste={onPaste}
        onChange={onChange}
        onBlur={onBlur}
        value={value}
        data-tid={rest['data-tid']}
      />
    </>
  )
}

NoraTextAreaInput.propTypes = {
  'data-tid': PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  labelCopy: PropTypes.string,
  labelClassName: PropTypes.string,
  textClassName: PropTypes.string,
  placeholder: PropTypes.string,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  onChange: PropTypes.func,
}
