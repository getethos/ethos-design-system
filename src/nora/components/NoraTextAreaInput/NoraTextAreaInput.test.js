import React from 'react'
import { NoraTextAreaInput } from './NoraTextAreaInput'
import renderer from 'react-test-renderer'

describe('NoraTextAreaInput', () => {
  it('default rendering', () => {
    const tree = renderer
      .create(
        <NoraTextAreaInput
          name="nombre"
          placeholder="enter text here"
          data-tid="text-nora-textareainput"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
