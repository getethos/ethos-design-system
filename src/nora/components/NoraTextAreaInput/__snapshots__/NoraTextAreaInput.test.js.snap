// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoraTextAreaInput default rendering 1`] = `
<textarea
  className="NoraTextInput "
  data-tid="text-nora-textareainput"
  name="nombre"
  placeholder="enter text here"
/>
`;
